driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryption25Lookup.java

```diff
@@ -0,0 +1,433 @@
+                client.getDatabase("db").getCollection("csfle").aggregate(pipeline).first());
+    }
+
+    @Test
+    public void case6() {
+        List<BsonDocument> pipeline = BsonArray.parse("[\n"
+                + "   {\"$match\" : {\"qe\" : \"qe\"}},\n"
+                + "   {\n"
+                + "      \"$lookup\" : {\n"
+                + "         \"from\" : \"qe2\",\n"
+                + "         \"as\" : \"matched\",\n"
+                + "         \"pipeline\" : [ {\"$match\" : {\"qe2\" : \"qe2\"}}, {\"$project\" : {\"_id\" : 0, \"__safeContent__\" : 0}} ]\n"
+                + "      }\n"
+                + "   },\n"
+                + "   {\"$project\" : {\"_id\" : 0, \"__safeContent__\" : 0}}\n"
+                + "]").stream().map(stage -> stage.asDocument()).collect(Collectors.toList());
+
+        assertEquals(
+                Document.parse("{\"qe\" : \"qe\", \"matched\" : [ {\"qe2\" : \"qe2\"} ]}"),
+                client.getDatabase("db").getCollection("qe").aggregate(pipeline).first());
+    }
+
+    @Test
+    public void case7() {
+        List<BsonDocument> pipeline = BsonArray.parse("[\n"
+                + "    {\"$match\" : {\"no_schema\" : \"no_schema\"}},\n"
+                + "    {\n"
+                + "        \"$lookup\" : {\n"
+                + "            \"from\" : \"no_schema2\",\n"
+                + "            \"as\" : \"matched\",\n"
+                + "            \"pipeline\" : [ {\"$match\" : {\"no_schema2\" : \"no_schema2\"}}, {\"$project\" : {\"_id\" : 0}} ]\n"
+                + "        }\n"
+                + "    },\n"
+                + "    {\"$project\" : {\"_id\" : 0}}\n"
+                + "]").stream().map(stage -> stage.asDocument()).collect(Collectors.toList());
+
+        assertEquals(
+                Document.parse("{\"no_schema\" : \"no_schema\", \"matched\" : [ {\"no_schema2\" : \"no_schema2\"} ]}"),
+                client.getDatabase("db").getCollection("no_schema").aggregate(pipeline).first());
+    }
+
+    @Test
+    public void case8() {
+        List<BsonDocument> pipeline = BsonArray.parse("[\n"
+                + "    {\"$match\" : {\"csfle\" : \"qe\"}},\n"
+                + "    {\n"
+                + "        \"$lookup\" : {\n"
+                + "            \"from\" : \"qe\",\n"
+                + "            \"as\" : \"matched\",\n"
+                + "            \"pipeline\" : [ {\"$match\" : {\"qe\" : \"qe\"}}, {\"$project\" : {\"_id\" : 0}} ]\n"
+                + "        }\n"
+                + "    },\n"
+                + "    {\"$project\" : {\"_id\" : 0}}\n"
+                + "]").stream().map(stage -> stage.asDocument()).collect(Collectors.toList());
+
+        assertCause(
+                MongoCryptException.class,
+                "not supported",
+                () -> client.getDatabase("db").getCollection("csfle").aggregate(pipeline).first());
+    }
+
+    @Test
+    public void case9() {
```

Comment:
We would need to skip this for versions earlier than 8.1.
```suggestion
    public void case9() {
    assumeTrue(serverVersionLessThan(8, 1));
```